File: server/Domain/Models/Meter.cs
using Domain.Enums;

namespace Domain.Models;

public class Meter
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public required string Location { get; set; }
    public string? MeterNumber { get; set; }
    public string? Addition { get; set; }
    public MeterType Type { get; set; }
    public IEnumerable<Reading>? Readings { get; set; }
    public IEnumerable<SharedMeter>? SharedMeters { get; set; }
    public User? User { get; set; }
}

File: client/src/app/api/models/meter.ts
import type { MeterType } from './meter-type';
import type { Reading } from './reading';

export interface Meter {
    id?: number;
    createDate?: string;
    updateDate?: (string) | (null);
    userId: number;
    location: (string) | (null);
    meterNumber?: (string) | (null);
    addition?: (string) | (null);
    type?: MeterType;
    readings?: ((Reading)[]) | (null);
  }
